mir_eval/display.py

```diff
@@ -0,0 +1,672 @@
+        idx = freqs > 0
+        freqs = freqs[idx]
+        voicings = voicings[idx]
+
+        if midi:
+            freqs = hz_to_midi(freqs)
+
+        n_voiced = sum(voicings)
+        voiced_times.extend([t] * n_voiced)
+        voiced_freqs.extend(freqs[voicings])
+        unvoiced_times.extend([t] * (len(freqs) - n_voiced))
+        unvoiced_freqs.extend(freqs[~voicings])
+
+    # Plot the voiced frequencies
+    ax.scatter(voiced_times, voiced_freqs, **style_voiced)
+
+    # Plot the unvoiced frequencies
+    if unvoiced:
+        ax.scatter(unvoiced_times, unvoiced_freqs, **style_unvoiced)
+
+    # Tick at integer midi notes
+    if midi:
+        ax.yaxis.set_minor_locator(MultipleLocator(1))
+    return ax
+
+
+def piano_roll(intervals, pitches=None, midi=None, **kwargs):
+    '''Plot a quantized piano roll as intervals
+
+    Parameters
+    ----------
+    intervals : np.ndarray, shape=(n, 2)
+        timing intervals for notes
+
+    pitches : np.ndarray, shape=(n,), optional
+        pitches of notes (in Hz).
+
+    midi : np.ndarray, shape=(n,), optional
+        pitches of notes (in MIDI numbers).
+
+        At least one of `pitches` or `midi` must be provided.
+
+    kwargs :
+        Additional keyword arguments to `labeled_intervals`.
+
+    Returns
+    -------
+    ax :
+        Handle to the plotting axis
+    '''
+
+    if midi is None:
+        midi = hz_to_midi(pitches)
+
+    scale = np.arange(128)
+    ax = labeled_intervals(intervals, np.round(midi).astype(int),
+                           label_set=scale,
+                           tick=False, **kwargs)
+
+    # Minor tick at each semitone
+    ax.yaxis.set_minor_locator(MultipleLocator(1))
+
+    ax.axis('auto')
+    return ax
+
+
+def separation(sources, fs=22050, labels=None, ax=None, **kwargs):
+    '''Source-separation visualization
+
+    Parameters
+    ----------
+    sources : list of np.ndarray
+        A list of waveform buffers corresponding to each source
+
+    fs : number > 0
+        The sampling rate
+
+    labels : list of strings
+        An optional list of descriptors corresponding to each source
+
+    ax : matplotlib.pyplot.axes
+        An axis handle on which to draw the intervals.
+        If none is provided, a new set of axes is created.
+
+    kwargs :
+        Additional keyword arguments to `scipy.signal.spectrogram`
+
+    Returns
+    -------
+    ax
+        The axis handle for this plot
+    '''
+    if ax is None:
+        ax = plt.gca()
+
+    if labels is None:
+        labels = ['Source {:d}'.format(_) for _ in range(len(sources))]
+
+    specs = []
+    cumspec = None
+    kwargs.setdefault('scaling', 'spectrum')
+
+    for i, src in enumerate(sources):
+        freqs, times, spec = spectrogram(src, fs=fs, **kwargs)
+        specs.append(spec)
+        if cumspec is None:
+            cumspec = spec.copy()
+        else:
+            cumspec += spec
+
+    ref_max = cumspec.max()
```

Comment:
If this is all you're using cumspec for, you should probably make a comment to this effect when fussing with it above.  (You could also just keep track of the max value seen in all spectrograms, which might simplify your code somewhat, but do whichever you prefer).